### ISSUE TITLE
Partition / Deprecate / Remove legacy logic from NdrSupport

### ISSUE THREAD

**Issue by joshpencheon** [PublicHealthEngland/ndr_support#3]
Some of the code in `NdrSupport` was extracted from legacy systems. It would not always be desirable to add this blindly to other projects.

Some of this logic has been encapsulated (e.g. `NdrSupport.attempt_date_patch!`), but in the current release is still active-by-default.

We should list methods that are undesirable (*resp.* unused by internal NDR projects!?), and put them through a deprecation cycle, after which they would be disabled-by-default (*resp.* removed entirely).

@timgentry, does this sound sensible?

**Comment by timgentry** [comment]
Definitely. Now that we have https://github.com/PublicHealthEngland/ndr_dev_support, I'm going to try and not use any of NdrSupport directly within our two new projects.

It's important to identify what https://github.com/PublicHealthEngland/ndr_import and our other gems use of it. Should RegexpRange move to NdrImport for example?
